packages/debug/src/browser/model/debug-thread.tsx

```diff
@@ -56,11 +56,18 @@ export class DebugThread extends DebugThreadData implements TreeElement {
         return this.session.id + ':' + this.raw.id;
     }
 
+    get threadId(): number {
+        return this.raw.id;
+    }
+
     protected _currentFrame: DebugStackFrame | undefined;
     get currentFrame(): DebugStackFrame | undefined {
         return this._currentFrame;
     }
     set currentFrame(frame: DebugStackFrame | undefined) {
+        if (this._currentFrame === frame) {
```

Comment:
Again, is there only ever one copy of the same frame?

Reply:
The code should update the existing one instead of creating new ones (see https://github.com/eclipse-theia/theia/blob/ade6c8de0fe4c223660fff67d1400d723ce89d3f/packages/debug/src/browser/model/debug-thread.tsx#L202). But comparing identifiers may be more robust indeed.

I pushed a new commit to address the issue.